asv_bench/benchmarks/strings.py

```diff
@@ -87,6 +84,29 @@ def time_repeat(self, repeats):
         self.s.str.repeat(self.repeat)
 
 
+class Cat(object):
+
+    goal_time = 0.2
+    params = ([None, 5], [None, ','], [None, '-'], [0.0, 1e-4, 0.1])
+    param_names = ['others', 'sep', 'na_rep', 'na_frac']
+
+    def setup(self, others, sep, na_rep, na_frac):
+        N = int(5e5)
+        mask_gen = lambda: np.random.choice([True, False], N,
+                                            p=[1 - na_frac, na_frac])
+        self.s = Series(tm.makeStringIndex(N)).where(mask_gen())
+        self.others = (DataFrame({i: tm.makeStringIndex(N).where(mask_gen())
+                                  for i in range(others)})
+                       if others is not None else None)
```

Comment:
Any reason you don't just use `0` instead of `None` as the parametrized argument? Would get rid of the need for a condition here

Reply:
Changed to `0` for consistency/legibility, but still need the condition, as `pd.DataFrame({'i' : ... for i in range(0)})` would yield a DataFrame (but `str.cat` needs `others=None` to self-concatenate).